File: PurrfectMatch.Web/src/app/models/review.ts
// Review Models matching C# DTOs

export interface ReviewDto {
  reviewId: number;
  userId: string;
  userName: string;
  shelterId: number;
  shelterName: string;
  rating: number;
  comment: string;
  createdAt: string;
}

export interface CreateReviewDto {
  shelterId: number;
  rating: number;
  comment: string;
}

export interface UpdateReviewDto {
  rating: number;
  comment: string;
}

export interface ShelterRatingStats {
  averageRating: number;
  count: number;
}

export interface ReviewFilterDto {
  shelterId?: number;
  userId?: string;
  minRating?: number;
  maxRating?: number;
  pageNumber?: number;
  pageSize?: number;
  sortBy?: string;
  sortDescending?: boolean;
}

File: PurrfectMatch.Domain/Entities/Review.cs



using System.ComponentModel.DataAnnotations;

namespace PurrfectMatch.Domain.Entities
{
    public partial class Review
    {
        [Key]
        public int ReviewId { get; set; }

        public string UserId { get; set; } = null!;

        public int ShelterId { get; set; }

        public int Rating { get; set; }

        public string? Comment { get; set; }

        public DateTime CreatedAt { get; set; }

        public virtual User User { get; set; } = null!;

        public virtual Shelter Shelter { get; set; } = null!;

        
    }
}


